### PR TITLE
Fix App.vue loading errors

### LINKED ISSUE #677: Refresh of assignmnent or class list results in 401 error
**Describe the bug**
As an instructor user when I refresh either the Assignment of Class List pages I get a 401 token required error

**To Reproduce**
1. Log onto a LMS as an instructor
2. Go to an EdEHR activity
3. Go to the Assignment page or the Class List page
4. Refresh your browser

**Expected behavior**
There should not be a 401 error

### LINKED ISSUE #681: Refresh browser for instructor fails
**Describe the bug**
When an instructor user first lands in the EdEHR the url includes the refresh token. But if the instructor refreshes the browser while this token is part of the url the page refresh fails. 

**To Reproduce**
1. Log into an LMS as an instructor and go to a course with an EdEHR activity and and launch it.
2. When you arrive on the course list page try a browser refresh.
3. Wait for at least a minute (the refresh timeout period)
4. Refresh the page. 
3. It fails with Token Expire alert

**Expected behavior**
1. The error message should not appear as an alert but use the system alert banner like other errors.
2. The refresh should use the previously retrieved token.

### PR DESCRIPTION
**Pull request by viniciusarre**
Closes #681 and closes #677

### CODE REVIEW

**Comment by viniciusarre** [comment]
@bryan-gilbert I have implemented the fix for the given issues, however, I have a couple of questions / concerns regarding #681.

1. The bug regarding the refreshToken occurred based on the order of the promise chain which I had created before. I have changed it now so that it, at first, looks for an authToken at first despite having a refreshToken or not. It fixed the given issue, however, there might be a context in which it could be problematic. For example, if we don't handle unsetting the authToken and the user tries to login with a different refreshToken while there's still an authToken present. Earlier we didn't need to worry too much about it, since it was designed with sessionStorage in mind, but this could be an issue with localStorage. I think we need to further discuss a potential lifecycle when the authToken can be safely unset.

2. The current alert banner was defined before the overall authToken/refreshToken logic was implemented. I have also used it for other error displaying. Could you please refer to the system alert banner which handles other errors? I think this could be potentially be turned into a separate issue, as there are other parts of the system which implements the same alert.

I look forward to further discussing these points with you.

Thank you!

File: client/src/App.vue [hunk 1]
```diff
@@ -49,14 +49,24 @@ export default {
                   return Promise.reject(Text.EXPIRED_REFRESH_TOKEN)
                 }
                 return StoreHelper.fetchTokenData(token, apiUrl)
+              }).catch(err => {
+                if (
+                  ( err.response.status === 401  && 
+                    err.response.data.includes('The token has expired.') &&
```
Comment (bryan-gilbert) [inline client/src/App.vue:55]: Using the expected error string here is fragile. If the message ever changes on the server this code will fail.  A less fragile approach needs two things.  First reduce the check to see if the work "expire" is in the message regardless of case. (eg. message.toLowerCase().includes('expired')) and second, add a comment to the server "the message MUST contain the word expired for clients to correctly handle a browser refresh when an expired refresh token is in the URL"

File: client/src/App.vue [hunk 2]
```diff
@@ -85,13 +95,20 @@ export default {
           EventBus.$emit(PAGE_DATA_REFRESH_EVENT)
         })
         .catch(err => {
-          alert(err + '\nSystem Error')
           StoreHelper.setLoading(null, false)
           this.$router.push('/')
+          setApiError(err + '\n. System Error')
```
Comment (bryan-gilbert) [inline client/src/App.vue:100]: Check that setApiError works with this newline. Does it convert the new line into an html break?

Comment (viniciusarre) [reply client/src/App.vue:100]: It doesn't. I will be removing this newline for now. Thank you for your observation!

**Pull request merged**